### ISSUE TITLE
Stats on the API?

### ISSUE THREAD

**Issue by ozguryuksek-ie** [buzzsprout/buzzsprout-api#34]
Hey team,

We are using Buzzsprout for our podcasts and we need to have Stats via API. I believe that this would be beneficial for you and your all customers. Is it possible to have this request on your roadmap?

**Comment by johnloringpollard** [comment]
Yea right now this is only available in the admin export. Have you seen that?

**Comment by ozguryuksek-ie** [comment]
I didn't see it. You didn't mean the download option on the stats page, right?

**Comment by johnloringpollard** [comment]
Yes, that option. If you have code that can log in and download the requests you could get them that way.

**Comment by chrisat3Seven** [comment]
there does not appear to be a way to download downloads per day, I can only view it in the web portal. Can I download this or access via API?

**Comment by TiagoFonseca** [comment]
Hi, is this something that you plan to do in the future?

**Comment by johnloringpollard** [comment]
This isn't on our list of top priorities. If this issue gets more traction we will be more likely to implement.

**Comment by ma-dtierney** [comment]
This would be helpful for my organization's internal dashboards.

**Comment by ma-dtierney** [comment]
Total downloads, and downloads per episode. If available from the API then we could update internal dashboards for our org. Thanks for the consideration.

**Comment by Zafiiovskiy** [comment]
Hi, just joining on the line that it would be awsome to have more complete stats in api. Like demographics, technologies, split by platfoms, etc. Everything that is availiable in Buzzsprout UI. Thanks! It's great to have the ability to display live stats for partners and donor at out website or other places

**Comment by deldrid1** [comment]
+1 for this - Its crazy to have to iterate every episode to calculate stats that are already being displayed by an endpoint for the frontend!

**Comment by jsovel** [comment]
+1 for for stats via the API.

**Comment by beeldrijmers** [comment]
Also looking to have more stats included in the API. The admin dashboards has all the nice statistics which I would very much like to display in my own dashboards. Would like to refrain from scraping the website to get this info, so a API would be appreciated!

**Comment by ljfoster97** [comment]
Another +1 for this! Would be hugely beneficial to avoid doing manual exports - especially since these now take a few days (!!!) to come through via email after they've been requested on the admin console. Alternatives like scraping via selenium are a pain to maintain or get other staff to maintain.

**Comment by caleb-bright** [comment]
I would also like all the stats displayed on Buzzsprout's admin dashboard available via the API. This would be very helpful information to be able to implement on a custom dashboard.

**Comment by pt3190** [comment]
Another +1 for stats to be included in the API, please!

**Comment by KSW1992** [comment]
+1 for stats to be available. I'd be really interested in being able to pull these through into dashboards.

**Comment by Joopsnijder** [comment]
+1

**Comment by Ben-Giant-VC** [comment]
It would be great to be able to pull the raw line items for the downloads so that we can cut and aggregate the data as we see fit on our end.

Some useful attributes would be: Episode, DSP, Location and Device

**Comment by wcarpenter110981** [comment]
+1 more, have been trying to figure this out for some time.  This would help all creators.

**Comment by jammcq** [comment]
I'd also love to see a stats api endpoint. I'd especially like to see when downloads are happening so I can create my own graphs showing download history per episode.
I could even settle for an export of the data. I know you have the data, you generate graphs.

**Comment by Joopsnijder** [comment]
+1. Would be very beneficial.

**Comment by iskayanir** [comment]
+1
It’s really necessary

**Comment by justinpage** [comment]
+1

**Comment by EM-SendLock** [comment]
+1  with a quick concrete use case to add some weight.

@johnloringpollard — thanks for staying engaged on this thread. Adding this, since you mentioned traction helps prioritize.

We run the Training That Works podcast and **just built a public "most-listened" leaderboard**
on our site (https://trainingthatworks.org/leaderboard). Right now the API only returns
cumulative `total_plays` per episode, so anything time-based has to come from manual
Stats → Overview CSV exports **we pull by hand**. We'd switch to a fully automatic API pull
the moment those stats were available.

**Keeping it simple:** the biggest win would just be **exposing the per-episode download stats**
you already show in the admin dashboard / Overview export through the API. Plus, if it's
easy, **whatever download history you already keep**. Nothing fancy needed to start; even the
numbers that are already in the existing export **would be a huge help**, and we're glad to
follow up with finer-grained requests later (detailed history, so we can track things like
how many weeks an episode has held #1).

**This is data you already generate**, it's for a public site, and we're happy to beta-test
anything. Thanks for considering!

**Comment by brydave** [comment]
Hey everyone, downloads are available via the API now.

Docs: https://github.com/buzzsprout/buzzsprout-api/blob/master/sections/stats.md

What’s there today:
- `GET /api/:podcast_id/downloads?date=YYYY-MM-DD` — episodes with downloads that day for quick checks (podcast-local date)
- `GET /api/:podcast_id/:episode_id/downloads?date=YYYY-MM-DD` — paginated download details (app, device, geo, media type) for more a more detailed breakdown.

These are single-date endpoints (no date range in one call). Episode detail is under `/api/:podcast_id/:episode_id/downloads`, not under `/episodes`. If you’re pulling a range, requests are limited to 60 per minute, but we may adjust as needed. We haven't changed the existing `total_plays`, so no changes there.

If you have any feedback or need something beyond what we just added, please open a new issue. That helps us prioritize these better than continuing this long thread.
